Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -326,6 +353,74 @@ theorem coe_map (f : E →L[𝕜] F) : (WeakSpace.map f : E → F) = f :=
   rfl
 #align weak_space.coe_map WeakSpace.coe_map
 
+/-- The canonical preimage in `E` of an open set in `WeakSpace 𝕜 E` is open in `E`. -/
+theorem isOpen_of_isOpen_WeakSpace (U : Set (WeakSpace 𝕜 E))
+    (hU : IsOpen[(WeakSpace.instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E))] U) :
+    IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U) := by
+  exact ((continuousLinearMapToWeakSpace 𝕜 E).cont).isOpen_preimage U hU
+
+/-- The canonical map from `WeakSpace 𝕜 E` to `E` is an open map. -/
+theorem isOpenMap_inv_toWeakSpace : IsOpenMap (toWeakSpace 𝕜 E).symm := by
+  apply IsOpenMap.of_inverse (continuousLinearMapToWeakSpace 𝕜 E).cont _ _
+  intro x
+  simp only [AddHom.toFun_eq_coe, LinearMap.coe_toAddHom, ContinuousLinearMap.coe_coe,
+    continuousLinearMapToWeakSpace_eq_toWeakSpace, LinearEquiv.symm_apply_apply]
+  intro x
+  simp only [AddHom.toFun_eq_coe, LinearMap.coe_toAddHom, ContinuousLinearMap.coe_coe,
+    continuousLinearMapToWeakSpace_eq_toWeakSpace, LinearEquiv.apply_symm_apply]
+
+/-- A set in `E` which is open in the weak topology is open. -/
+theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
+    (hV : IsOpen[(WeakSpace.instTopologicalSpace)]
+    ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
```

Comment:
Just like the preimage version that follows from continuity, I'm not sure we need this as it's just the `simp` of the application of the `IsOpenMap` result. That is, the following proves this result:
```
theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
    (hV : IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
  simpa [Set.image_image] using isOpenMap_inv_toWeakSpace _ hV
```
However, we can keep it (without the `'` in the name) if you prefer.
Note also, you don't need to keep specifying `IsOpen[topologicalSpace-instance]`, you can just use `IsOpen`. That's one of the reasons we have a type synonym.

Reply:
thanks, I used the code you suggested.